### PR TITLE
test: allow gc to get unlucky

### LINKED ISSUE #3824: test_pageserver_restarts_under_worload is flaky
```
2023-03-15T11:05:41.4093101Z ==================================== ERRORS ====================================
2023-03-15T11:05:41.4094222Z _________ ERROR at teardown of test_pageserver_restarts_under_worload __________
2023-03-15T11:05:41.4094977Z [gw14] linux -- Python 3.9.2 /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/bin/python
2023-03-15T11:05:41.4095822Z /github/home/.cache/pypoetry/virtualenvs/neon-_pxWMzVK-py3.9/lib/python3.9/site-packages/allure_commons/_allure.py:200: in __call__
2023-03-15T11:05:41.4096290Z     return self._fixture_function(*args, **kwargs)
2023-03-15T11:05:41.4096590Z test_runner/fixtures/neon_fixtures.py:1056: in _shared_simple_env
2023-03-15T11:05:41.4096936Z     yield env
2023-03-15T11:05:41.4097181Z test_runner/fixtures/neon_fixtures.py:853: in __exit__
2023-03-15T11:05:41.4097475Z     self.env.pageserver.assert_no_errors()
2023-03-15T11:05:41.4098351Z test_runner/fixtures/neon_fixtures.py:2172: in assert_no_errors
2023-03-15T11:05:41.4098664Z     assert not errors
2023-03-15T11:05:41.4099250Z E   AssertionError: assert not ['2023-03-15T11:02:36.952113Z ERROR gc_loop{tenant_id=9ad7798ee55e644ea65b8a7d9a353175}: Gc failed, retrying in 2s: Cannot run GC iteration on inactive tenant\n']
```

https://neon-github-public-dev.s3.amazonaws.com/reports/pr-3821/release/4425498523/index.html#suites/542b1248464b42cc5a4560f408115965/a5ad46f5e6569269/

### PR DESCRIPTION
**Pull request by koivunej**
this failure case was probably introduced by b220ba6, because earlier the gc would always have run fast enough for restart every 1s. however, test got added later, so we have just been lucky.

fixes #3824 by allowing this error to happen.

### CODE REVIEW

File: test_runner/regress/test_pageserver_restarts_under_workload.py [hunk 1]
```diff
@@ -17,6 +17,12 @@ def test_pageserver_restarts_under_worload(neon_simple_env: NeonEnv, pg_bin: PgB
     n_restarts = 10
     scale = 10
 
+    # the background task may complete the init task delay after finding an
+    # active tenant, but shutdown starts right before Tenant::gc_iteration
+    env.pageserver.allowed_errors.append(
+        r"Gc failed, retrying in \S+: Cannot run GC iteration on inactive tenant"
```
Comment (koivunej) [inline test_runner/regress/test_pageserver_restarts_under_workload.py:23]: This isn't enough, we do `re.match` not search.

Comment (koivunej) [reply test_runner/regress/test_pageserver_restarts_under_workload.py]: [8dda5a7](https://github.com/neondatabase/neon/pull/3826/commits/8dda5a784bb38c27c49300d6155f1d384eec1079)

**Comment by koivunej** [comment]
No reviews yet. I am quite certain we can see from pre-random_init_delay test runs that the PR description is correct. Will look one up later.

**Review by knizhnik — APPROVED** [review]

**Review by hlinnaka — APPROVED** [review]

**Comment by koivunej** [comment]
Affected test was added *after* `random_init_delay` so I conclude we have just been lucky. I don't think the test actually completes in `1s * N` (`-T` parameter to pgbench, assuming it does what I think it does) so perhaps it should be adjusted, but this is not critical because we will have at least 1-2 rounds of restarts while pgbench is active even if few last are done without any load.

**Pull request merged**